packages/babel-core/src/transformation/file/file.js

```diff
@@ -149,15 +151,26 @@ export default class File {
       node._compact = true;
     });
 
-    this.path.unshiftContainer("body", nodes);
+    return { declaration: nodes, reference: uid };
+  }
+
+  addHelper(name: string): Object {
+    const { declaration, reference } = this._loadHelper(
```

Comment:
It seems like it has gotten pretty complicated in here.

> This commit updates the getDependency callback passed to helpers.get(). That callback only used to return an Identifier or MemberExpression which represents the helper; now it can return both the reference and the code of the dependency

This is the primary reason for these changes, is that right?

Having the `helpers.get` function return the dependencies too seems like it adds a lot of complexity to this logic and I'm not sure what is driving that change. Could you expand on that to help the discussion?

I'm wondering if we'd be better off changing the helper API into two parts, along the lines of a `.getGraph(name: string)` that would return a list of all the dependencies with the expectation that if you need to insert the dependencies, you'd do

```
helpers.getGraph(name).forEach(helper => helpers.get(...
```

That way generating an individual helper can just assume that your `getDependency` callback will always point at something that has already been inserted.

Thoughts?

Reply:
It seems like I have the tendency to make things way more complex than needed 😅 
While writing an answer to your comment I realized that there was no reason to make the `helpers.get` function inject the dependencies in the helper.